pkgs/development/interpreters/python/tests.nix

```diff
@@ -101,6 +101,24 @@ let
     };
   };
 
+  # Tests to ensure overriding works as expected.
+  overrideTests = let
+    extension = self: super: {
+      foobar = super.numpy;
+    };
+  in {
+    test-packageOverrides = let
+      myPython = let
+        self = python.override {
+          packageOverrides = extension;
+          inherit self;
+        };
+      in self;
+    in assert myPython.pkgs.foobar == myPython.pkgs.numpy; myPython.withPackages(ps: with ps; [ foobar ]);
+    # overrideScope is broken currently
+    # test-overrideScope = let
```

Comment:
@Ericson2314 @Infinisil overrideScope seems to function partially here; the package set is correctly overridden, however, the interpreter that is part of the set is not informed of the update, and hence `withPackages` or `buildEnv` is given the old set. What's needed I think here is what is done with `packageOverrides`; to pass `self` (the new interpreter) as well.

Ideally we would not need to pass the interpreter when overriding, but I don't see any way around that when `withPackages`/`buildEnv` is part of the interpreter, and not of the package set.

Any views on that? This relates to the `pythonPackages` versus `python.pkgs` and where should `withPackages` be discussion on the RFC.